arch/xtensa/src/esp32/esp32_allocateheap.c

```diff
@@ -130,6 +143,10 @@
 void up_allocate_heap(FAR void **heap_start, size_t *heap_size)
 {
   board_autoled_on(LED_HEAPALLOCATE);
+#ifdef CONFIG_XTENSA_IMEM_MAXIMIZE_HEAP_REGION
+  *heap_size = HEAP_REGION_OFFSET;
+  *heap_start = (size_t)(HEAP_REGION1_END - *heap_size);
```

Comment:
Hi @yamt  Just to reserve part of the heap memory of `region 1` for the NuttX system, because I found that if `region 1` is all used as the internal heap, there will be issue with the NuttX initialization or Wi-Fi connection process.

Reply:
my question was "(size_t)". if it was "(FAR void *)", i can understand.
does this even compile without warnings?

Reply:
Got it, I have modified.